cvat/apps/quality_control/models.py

```diff
@@ -222,4 +310,26 @@ def to_dict(self):
 
     @property
     def organization_id(self):
-        return getattr(self.task.organization, "id", None)
+        if self.task:
+            return getattr(self.task.organization, "id", None)
+        elif self.project:
+            return getattr(self.project.organization, "id", None)
+        return None
+
+    @transaction.atomic
+    def save(self, *args, **kwargs) -> None:
+        try:
+            return super().save(*args, **kwargs)
+        except IntegrityError as ex:
+            message = str(ex)
+            if "duplicate key value violates unique constraint" in message:
```

Comment:
This doesn't seem very robust. I think It would be better to query the database to see if the given quality parameters really already exist.

Reply:
The error should come from the DB, what do you mean?

Reply:
I mean that you have no guarantee that the error will contain this wording in the future.

Reply:
If it changes in future, it should be catched by a corresponding test.

Reply:
Is there a corresponding test? I couldn't find one.

Reply:
Added a dedicated test for this, but we don't have such tests in general.